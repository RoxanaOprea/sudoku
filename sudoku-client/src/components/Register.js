import React from "react";
import TextField from "@material-ui/core/TextField";
import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import { Link } from "react-router-dom";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap",
    marginTop: "150px",
    width: "300px",
    margin: "auto"
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit
  },
  button: {
    margin: "65px"
  },
  input: {
    display: "none"
  },
  smallContainer: {
    width: "200px",
    margin: "auto"
  },
  link: {
    marginRight: "50px"
  }
});

class Register extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: null,
      password: null,
      newUsers: []
    };
  }

  handleChange(event, inputName) {
    this.setState({ [inputName]: event.target.value });
  }

  handleSubmit = event => {
    event.preventDefault();

    fetch("http://localhost:3001/register", {
      method: "POST",
      body: JSON.stringify({
        email: this.state.email,
        password: this.state.password
      }),
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json"
      }
    })
      .then(response => {
        return response.json();
      })
      .then(res => {
        const users = this.state.newUsers;
        users.push({ email: res.email, password: res.password });

        this.setState({ newUsers: users });
      })
      .catch(err => console.log(err));
  };

  render() {
    const { classes } = this.props;
    console.log(this.state.newUsers);

    return (
      <div>
        <MuiThemeProvider>
          <AppBar title="Registration">
            <Toolbar>
              <Typography variant="h6" color="inherit">
                Registration Page
              </Typography>
            </Toolbar>
          </AppBar>
          <div>
            <form
              className={classes.container}
              noValidate
              autoComplete="off"
              onSubmit={this.handleSubmit}
            >
              <TextField
                id="register-email-input"
                label="Email"
                className={classes.textField}
                type="email"
                name="email"
                autoComplete="email"
                margin="normal"
                variant="outlined"
                onChange={event => this.handleChange(event, "email")}
              />
              <TextField
                id="register-password-input"
                label="Password"
                className={classes.textField}
                type="password"
                autoComplete="current-password"
                margin="normal"
                variant="outlined"
                onChange={event => this.handleChange(event, "password")}
              />
              <Button
                type="submit"
                variant="outlined"
                color="primary"
                className={classes.button}
              >
                Register
              </Button>
            </form>
            <div className={classes.smallContainer}>
              <Link to="/" className={classes.link}>
                Back to Login Page
              </Link>
            </div>
          </div>
        </MuiThemeProvider>
      </div>
    );
  }
}

Register.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Register);
