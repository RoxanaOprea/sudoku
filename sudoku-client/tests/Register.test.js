import React from "react";
import { shallow } from "enzyme";
import Register from "../src/components/Register";
import toJson from "enzyme-to-json";
import Link from "react-router-dom";
import { render } from "react-dom";

jest.mock("react-dom");

let component = null;

describe("Register component", () => {
  beforeEach(() => {
    component = shallow(<Register />);
  });

  it("renders correctly", () => {
    expect(component).toMatchSnapshot();
    expect(component.dive()).toMatchSnapshot();
  });

  it("should have a `form` element", () => {
    expect(component.dive().find("form").length).toBe(1);
  });

  it("renders a text field for email", () => {
    expect(component.dive().find("#register-email-input").length).toEqual(1);
  });

  it("renders a text field for password", () => {
    expect(component.dive().find("#register-password-input").length).toEqual(1);
  });

  it("renders a button", () => {
    expect(component.dive().find("#register-button").length).toEqual(1);
  });

  it("renders a link", () => {
    expect(component.dive().find("#login-page").length).toEqual(1);
  });
});

describe("Register component actions", () => {
  beforeEach(() => {
    component = shallow(<Register />).dive();
  });

  it("should respond to change event and change the state for email field", () => {
    const emailInput = component.find("#register-email-input");
    emailInput.first().simulate("change", {
      target: { name: "email", value: "test@test.com" }
    });

    expect(component.instance().state.email).toEqual("test@test.com");
  });

  it("should respond to change event and change the state for password field", () => {
    const passwordInput = component.find("#register-password-input");
    passwordInput.first().simulate("change", {
      target: { name: "password", value: "test" }
    });

    expect(component.instance().state.password).toEqual("test");

    //test link component
    // beforeEach(() => {
    //   window.fetch = jest.fn().mockImplementation(()=> {
    //     Promise.resolve({})
    //   })
    // })
  });
});
