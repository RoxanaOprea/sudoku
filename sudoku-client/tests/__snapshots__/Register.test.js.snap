// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Register) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Register-button-3",
        "container": "Register-container-1",
        "input": "Register-input-4",
        "link": "Register-link-6",
        "smallContainer": "Register-smallContainer-5",
        "textField": "Register-textField-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Register-button-3",
          "container": "Register-container-1",
          "input": "Register-input-4",
          "link": "Register-link-6",
          "smallContainer": "Register-smallContainer-5",
          "textField": "Register-textField-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Register component renders correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    classes={
      Object {
        "button": "Register-button-3",
        "container": "Register-container-1",
        "input": "Register-input-4",
        "link": "Register-link-6",
        "smallContainer": "Register-smallContainer-5",
        "textField": "Register-textField-2",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MuiThemeProvider
        theme={Object {}}
      >
        <WithStyles(AppBar)
          title="Registration"
        >
          <WithStyles(Toolbar)>
            <WithStyles(Typography)
              color="inherit"
              variant="h6"
            >
              Registration Page
            </WithStyles(Typography)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>
        <div>
          <form
            autoComplete="off"
            className="Register-container-1"
            noValidate={true}
            onSubmit={[Function]}
          >
            <TextField
              autoComplete="email"
              className="Register-textField-2"
              id="register-email-input"
              label="Email"
              margin="normal"
              name="email"
              onChange={[Function]}
              required={false}
              select={false}
              type="email"
              variant="outlined"
            />
            <TextField
              autoComplete="current-password"
              className="Register-textField-2"
              id="register-password-input"
              label="Password"
              margin="normal"
              name="password"
              onChange={[Function]}
              required={false}
              select={false}
              type="password"
              variant="outlined"
            />
            <WithStyles(Button)
              className="Register-button-3"
              color="primary"
              id="register-button"
              type="submit"
              variant="outlined"
            >
              Register
            </WithStyles(Button)>
          </form>
          <div
            className="Register-smallContainer-5"
          >
            <Link
              className="Register-link-6"
              id="login-page"
              replace={false}
              title="Login"
              to="/"
            >
              Back to Login Page
            </Link>
          </div>
        </div>
      </MuiThemeProvider>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)
            title="Registration"
          >
            <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Registration Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>,
          <div>
            <form
              autoComplete="off"
              className="Register-container-1"
              noValidate={true}
              onSubmit={[Function]}
            >
              <TextField
                autoComplete="email"
                className="Register-textField-2"
                id="register-email-input"
                label="Email"
                margin="normal"
                name="email"
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                variant="outlined"
              />
              <TextField
                autoComplete="current-password"
                className="Register-textField-2"
                id="register-password-input"
                label="Password"
                margin="normal"
                name="password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="outlined"
              />
              <WithStyles(Button)
                className="Register-button-3"
                color="primary"
                id="register-button"
                type="submit"
                variant="outlined"
              >
                Register
              </WithStyles(Button)>
            </form>
            <div
              className="Register-smallContainer-5"
            >
              <Link
                className="Register-link-6"
                id="login-page"
                replace={false}
                title="Login"
                to="/"
              >
                Back to Login Page
              </Link>
            </div>
          </div>,
        ],
        "theme": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Registration Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>,
            "title": "Registration",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Registration Page
              </WithStyles(Typography)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Registration Page",
                "color": "inherit",
                "variant": "h6",
              },
              "ref": null,
              "rendered": "Registration Page",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                autoComplete="off"
                className="Register-container-1"
                noValidate={true}
                onSubmit={[Function]}
              >
                <TextField
                  autoComplete="email"
                  className="Register-textField-2"
                  id="register-email-input"
                  label="Email"
                  margin="normal"
                  name="email"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="email"
                  variant="outlined"
                />
                <TextField
                  autoComplete="current-password"
                  className="Register-textField-2"
                  id="register-password-input"
                  label="Password"
                  margin="normal"
                  name="password"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  variant="outlined"
                />
                <WithStyles(Button)
                  className="Register-button-3"
                  color="primary"
                  id="register-button"
                  type="submit"
                  variant="outlined"
                >
                  Register
                </WithStyles(Button)>
              </form>,
              <div
                className="Register-smallContainer-5"
              >
                <Link
                  className="Register-link-6"
                  id="login-page"
                  replace={false}
                  title="Login"
                  to="/"
                >
                  Back to Login Page
                </Link>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "children": Array [
                  <TextField
                    autoComplete="email"
                    className="Register-textField-2"
                    id="register-email-input"
                    label="Email"
                    margin="normal"
                    name="email"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="email"
                    variant="outlined"
                  />,
                  <TextField
                    autoComplete="current-password"
                    className="Register-textField-2"
                    id="register-password-input"
                    label="Password"
                    margin="normal"
                    name="password"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="password"
                    variant="outlined"
                  />,
                  <WithStyles(Button)
                    className="Register-button-3"
                    color="primary"
                    id="register-button"
                    type="submit"
                    variant="outlined"
                  >
                    Register
                  </WithStyles(Button)>,
                ],
                "className": "Register-container-1",
                "noValidate": true,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoComplete": "email",
                    "className": "Register-textField-2",
                    "id": "register-email-input",
                    "label": "Email",
                    "margin": "normal",
                    "name": "email",
                    "onChange": [Function],
                    "required": false,
                    "select": false,
                    "type": "email",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoComplete": "current-password",
                    "className": "Register-textField-2",
                    "id": "register-password-input",
                    "label": "Password",
                    "margin": "normal",
                    "name": "password",
                    "onChange": [Function],
                    "required": false,
                    "select": false,
                    "type": "password",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Register",
                    "className": "Register-button-3",
                    "color": "primary",
                    "id": "register-button",
                    "type": "submit",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": "Register",
                  "type": [Function],
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="Register-link-6"
                  id="login-page"
                  replace={false}
                  title="Login"
                  to="/"
                >
                  Back to Login Page
                </Link>,
                "className": "Register-smallContainer-5",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Back to Login Page",
                  "className": "Register-link-6",
                  "id": "login-page",
                  "replace": false,
                  "title": "Login",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Back to Login Page",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MuiThemeProvider
          theme={Object {}}
        >
          <WithStyles(AppBar)
            title="Registration"
          >
            <WithStyles(Toolbar)>
              <WithStyles(Typography)
                color="inherit"
                variant="h6"
              >
                Registration Page
              </WithStyles(Typography)>
            </WithStyles(Toolbar)>
          </WithStyles(AppBar)>
          <div>
            <form
              autoComplete="off"
              className="Register-container-1"
              noValidate={true}
              onSubmit={[Function]}
            >
              <TextField
                autoComplete="email"
                className="Register-textField-2"
                id="register-email-input"
                label="Email"
                margin="normal"
                name="email"
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                variant="outlined"
              />
              <TextField
                autoComplete="current-password"
                className="Register-textField-2"
                id="register-password-input"
                label="Password"
                margin="normal"
                name="password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                variant="outlined"
              />
              <WithStyles(Button)
                className="Register-button-3"
                color="primary"
                id="register-button"
                type="submit"
                variant="outlined"
              >
                Register
              </WithStyles(Button)>
            </form>
            <div
              className="Register-smallContainer-5"
            >
              <Link
                className="Register-link-6"
                id="login-page"
                replace={false}
                title="Login"
                to="/"
              >
                Back to Login Page
              </Link>
            </div>
          </div>
        </MuiThemeProvider>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(AppBar)
              title="Registration"
            >
              <WithStyles(Toolbar)>
                <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Registration Page
                </WithStyles(Typography)>
              </WithStyles(Toolbar)>
            </WithStyles(AppBar)>,
            <div>
              <form
                autoComplete="off"
                className="Register-container-1"
                noValidate={true}
                onSubmit={[Function]}
              >
                <TextField
                  autoComplete="email"
                  className="Register-textField-2"
                  id="register-email-input"
                  label="Email"
                  margin="normal"
                  name="email"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="email"
                  variant="outlined"
                />
                <TextField
                  autoComplete="current-password"
                  className="Register-textField-2"
                  id="register-password-input"
                  label="Password"
                  margin="normal"
                  name="password"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  variant="outlined"
                />
                <WithStyles(Button)
                  className="Register-button-3"
                  color="primary"
                  id="register-button"
                  type="submit"
                  variant="outlined"
                >
                  Register
                </WithStyles(Button)>
              </form>
              <div
                className="Register-smallContainer-5"
              >
                <Link
                  className="Register-link-6"
                  id="login-page"
                  replace={false}
                  title="Login"
                  to="/"
                >
                  Back to Login Page
                </Link>
              </div>
            </div>,
          ],
          "theme": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Toolbar)>
                <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Registration Page
                </WithStyles(Typography)>
              </WithStyles(Toolbar)>,
              "title": "Registration",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Typography)
                  color="inherit"
                  variant="h6"
                >
                  Registration Page
                </WithStyles(Typography)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Registration Page",
                  "color": "inherit",
                  "variant": "h6",
                },
                "ref": null,
                "rendered": "Registration Page",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  autoComplete="off"
                  className="Register-container-1"
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <TextField
                    autoComplete="email"
                    className="Register-textField-2"
                    id="register-email-input"
                    label="Email"
                    margin="normal"
                    name="email"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="email"
                    variant="outlined"
                  />
                  <TextField
                    autoComplete="current-password"
                    className="Register-textField-2"
                    id="register-password-input"
                    label="Password"
                    margin="normal"
                    name="password"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="password"
                    variant="outlined"
                  />
                  <WithStyles(Button)
                    className="Register-button-3"
                    color="primary"
                    id="register-button"
                    type="submit"
                    variant="outlined"
                  >
                    Register
                  </WithStyles(Button)>
                </form>,
                <div
                  className="Register-smallContainer-5"
                >
                  <Link
                    className="Register-link-6"
                    id="login-page"
                    replace={false}
                    title="Login"
                    to="/"
                  >
                    Back to Login Page
                  </Link>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "children": Array [
                    <TextField
                      autoComplete="email"
                      className="Register-textField-2"
                      id="register-email-input"
                      label="Email"
                      margin="normal"
                      name="email"
                      onChange={[Function]}
                      required={false}
                      select={false}
                      type="email"
                      variant="outlined"
                    />,
                    <TextField
                      autoComplete="current-password"
                      className="Register-textField-2"
                      id="register-password-input"
                      label="Password"
                      margin="normal"
                      name="password"
                      onChange={[Function]}
                      required={false}
                      select={false}
                      type="password"
                      variant="outlined"
                    />,
                    <WithStyles(Button)
                      className="Register-button-3"
                      color="primary"
                      id="register-button"
                      type="submit"
                      variant="outlined"
                    >
                      Register
                    </WithStyles(Button)>,
                  ],
                  "className": "Register-container-1",
                  "noValidate": true,
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "autoComplete": "email",
                      "className": "Register-textField-2",
                      "id": "register-email-input",
                      "label": "Email",
                      "margin": "normal",
                      "name": "email",
                      "onChange": [Function],
                      "required": false,
                      "select": false,
                      "type": "email",
                      "variant": "outlined",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "autoComplete": "current-password",
                      "className": "Register-textField-2",
                      "id": "register-password-input",
                      "label": "Password",
                      "margin": "normal",
                      "name": "password",
                      "onChange": [Function],
                      "required": false,
                      "select": false,
                      "type": "password",
                      "variant": "outlined",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Register",
                      "className": "Register-button-3",
                      "color": "primary",
                      "id": "register-button",
                      "type": "submit",
                      "variant": "outlined",
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="Register-link-6"
                    id="login-page"
                    replace={false}
                    title="Login"
                    to="/"
                  >
                    Back to Login Page
                  </Link>,
                  "className": "Register-smallContainer-5",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Back to Login Page",
                    "className": "Register-link-6",
                    "id": "login-page",
                    "replace": false,
                    "title": "Login",
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Back to Login Page",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
